src/libraries/Common/src/Interop/Linux/procfs/Interop.ProcFsStat.TryReadStatusFile.cs

```diff
@@ -149,21 +150,46 @@ internal static bool TryParseStatusFile(string statusFilePath, out ParsedStatus
             return true;
         }
 
-        private static bool TryReadFile(string filePath, ReusableTextReader reusableReader, [NotNullWhen(true)] out string? fileContents)
+        private static bool TryReadFile(string path, [NotNullWhen(true)] out string? contents)
         {
+            Debug.Assert(!string.IsNullOrEmpty(path));
+
+            byte[] bytes = ArrayPool<byte>.Shared.Rent(4096);
+            int count = 0;
+
             try
             {
-                using (var fileStream = new FileStream(filePath, FileMode.Open, FileAccess.Read, FileShare.Read, bufferSize: 1, useAsync: false))
+                using var fileStream = new FileStream(path, FileMode.Open, FileAccess.Read, FileShare.Read, bufferSize: 1, useAsync: false);
+
+                while (true)
                 {
-                    fileContents = reusableReader.ReadAllText(fileStream);
-                    return true;
+                    int read = fileStream.Read(bytes, count, bytes.Length - count);
+                    if (read == 0)
+                    {
+                        contents = Encoding.UTF8.GetString(bytes, 0, count);
+                        return true;
+                    }
+
+                    count += read;
+                    if (count >= bytes.Length)
+                    {
+                        byte[] temp = ArrayPool<byte>.Shared.Rent(bytes.Length * 2);
+                        Array.Copy(bytes, temp, bytes.Length);
+                        byte[] toReturn = bytes;
+                        bytes = temp;
+                        ArrayPool<byte>.Shared.Return(toReturn);
+                    }
                 }
             }
             catch (IOException)
             {
-                fileContents = null;
+                contents = null;
                 return false;
             }
+            finally
+            {
+                ArrayPool<byte>.Shared.Return(bytes);
```

Comment:
@jkotas I have seen in some cases we don't bother to return in exceptional cases. Yet, I could imagine for some OS reason, we are always failing to read this file, and we are called often, so we drain the array pool. I guess it depends on the context?

Reply:
Yep, it depends on the context. I think the try/finally is appropriate here.